umbraco/Umbraco-CMS
# Migrate authorization filters to authorization policies

Migrate authorization filters to authorization policies. Not all implementations of IAuthorizationFilter have been replaced since not all are necessary to be migrated such as InstallAuthorizeAttribute. But all filters that deal with back office authorization have been migrated.

This creates a different policy (name) per access rule. This does mean there are a lot of rules listed in the AuthorizationPolicies but it does mean we also create all of the rules up front. There is an alternative to this which would be to revert back to custom IActionFilter attributes that uses the IAuthorizationService directly and invoke custom authorization policies via [resource authorization](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/resourcebased?view=aspnetcore-5.0) and dynamically pass in the requirements/resources based on the parameters specified in those attributes. Basically it's either we define the rules/policies in one place, or they are dispersed within attributes in controllers. My vote is to leave them all declared as policies like in this PR since we can then target them just by name and they are managed globally.

## Testing

* Ensure the back office works as expected
* Create user(s) that have different access levels and ensure authorization works as expected
* There's a lot of scenarios, not expecting that all are tested - this is mostly code review since the code for the authorization itself hasn't changed
